openmls/src/group/tests/test_external_commit_validation.rs

```diff
@@ -4,152 +4,83 @@
-        .build();
-
-    // === Alice creates a group ===
-    let alice_group = MlsGroup::new_with_group_id(
-        backend,
-        &alice_credential.signer,
-        &mls_group_config,
-        group_id,
-        alice_credential.credential_with_key.clone(),
-    )
-    .expect("An unexpected error occurred.");
-
-    // Bob wants to commit externally.
-
-    // Have Alice export everything that bob needs.
-    let verifiable_group_info = alice_group
-        .export_group_info(backend, &alice_credential.signer, false)
-        .unwrap()
-        .into_group_info()
-        .unwrap();
-    let tree_option = alice_group.export_ratchet_tree();
-
-    let (_bob_group, message) = MlsGroup::join_by_external_commit(
-        backend,
-        &bob_credential.signer,
-        Some(&tree_option),
-        verifiable_group_info,
-        alice_group.configuration(),
-        &[],
-        bob_credential.credential_with_key.clone(),
-    )
-    .expect("Error initializing group externally.");
-
-    let serialized_message = message
-        .tls_serialize_detached()
-        .expect("Could not serialize message.");
-
-    let message = MlsMessageIn::tls_deserialize(&mut serialized_message.as_slice())
-        .expect("Could not deserialize message.")
-        .into_plaintext()
-        .expect("Message was not a plaintext.");
-
-    assert!(matches!(message.sender(), Sender::NewMemberCommit));
-
-    let original_plaintext = message.clone();
-
-    ECValidationTestSetup {
-        alice_group,
-        alice_credential: Box::new(alice_credential),
-        bob_credential: Box::new(bob_credential),
-        plaintext: message,
-        original_plaintext,
-    }
-}
-
 // ValSem240: External Commit, inline Proposals: There MUST be at least one ExternalInit proposal.
 #[apply(ciphersuites_and_backends)]
 fn test_valsem240(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
-    // Test with PublicMessage
     let ECValidationTestSetup {
         mut alice_group,
-        alice_credential: _,
         bob_credential,
-        mut plaintext,
-        original_plaintext,
+        public_message_commit,
+        ..
     } = validation_test_setup(PURE_PLAINTEXT_WIRE_FORMAT_POLICY, ciphersuite, backend);
 
-    let mut content = if let FramedContentBody::Commit(commit) = plaintext.content() {
-        commit.clone()
-    } else {
-        panic!("Unexpected content type.");
-    };
-
-    // Remove the external init proposal in the commit.
-    let proposal_position = content
-        .proposals
-        .iter()
-        .position(|proposal| match proposal {
-            ProposalOrRef::Proposal(proposal) => proposal.is_type(ProposalType::ExternalInit),
-            ProposalOrRef::Reference(_) => false,
-        })
-        .expect("Couldn't find external init proposal.");
-
-    content.proposals.remove(proposal_position);
-
-    plaintext.set_content(FramedContentBody::Commit(content));
+    // Setup
+    let public_message_commit_bad = {
```

Comment:
Narrowed scoping. This makes it a lot easier to see what is happening in the test. The intention here was to better see which parts could be refactored out as functions.